pkg/appfile/helm/helm.go

```diff
@@ -0,0 +1,93 @@
+package helm
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	helmapi "github.com/oam-dev/kubevela/pkg/appfile/helm/flux2apis"
+)
+
+// GenerateHelmReleaseAndHelmRepo constructs HelmRelease and HelmRepository in unstructured format
+func GenerateHelmReleaseAndHelmRepo(helmSpec *v1alpha2.Helm, svcName, appName, ns string, values map[string]interface{}) (helmRls, helmRepo *unstructured.Unstructured, err error) {
+	defaultIntervalDuration := &metav1.Duration{Duration: 5 * time.Minute}
+
+	rlsSpec := &helmapi.HelmReleaseSpec{}
+	repoSpec := &helmapi.HelmRepositorySpec{}
+	if err := json.Unmarshal(helmSpec.Release.Raw, rlsSpec); err != nil {
+		return nil, nil, err
+	}
+	if err := json.Unmarshal(helmSpec.Repository.Raw, repoSpec); err != nil {
+		return nil, nil, err
+	}
+
+	// construct HelmRepository data
+	helmRepo = &unstructured.Unstructured{}
+	helmRepo.SetGroupVersionKind(helmapi.HelmRepositoryGVK)
+	helmRepo.SetNamespace(ns)
+	repoName := fmt.Sprintf("%s-%s", appName, svcName)
+	helmRepo.SetName(repoName)
+
+	if repoSpec.Interval == nil {
+		repoSpec.Interval = defaultIntervalDuration
```

Comment:
what does the interval mean?

Reply:
It means the interval that helm-controller reconcile the helm release. It's a required field for flux/helm-release.